cpp/include/cudf/detail/aggregation/aggregation.hpp

```diff
@@ -960,6 +964,47 @@ class udf_aggregation final : public rolling_aggregation {
   }
 };
 
+/**
+ * @brief Derived class for specifying a custom aggregation specified in host-based UDF.
+ */
+class host_udf_aggregation final : public groupby_aggregation {
```

Comment:
Consider moving this to a separate header as well.
And define the non-templated functions in a .cpp file.

Reply:
Done, moving the functions calling `udf_ptr` into a separate source file.